service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -180,7 +182,7 @@ private String buildNarrativeStatementForDiagnosticReport(DiagnosticReport diagn
         var narrativeStatementTemplateParameters = NarrativeStatementTemplateParameters.builder()
             .narrativeStatementId(randomIdGeneratorService.createNewId())
             .commentType(commentType)
-            .issuedDate(DateFormatUtil.toHl7Format(diagnosticReport.getIssued().toInstant()))
+            .commentDate(DateFormatUtil.toHl7Format(diagnosticReport.getIssued().toInstant()))
```

Comment:
We can't use .toInstant() from the HAPI FHIR library because of variable-precision dates/time e.g. if there is only a date we can't assume the time is midnight. There is another DateFormatUtil method that support the FHIR *Element class instead.